### PR TITLE
Save Customized Block in LePatron

### PR DESCRIPTION
**Pull request by omar-bear**
### Description:
The primary goal of this feature is to allow users of LePatron to save an existing block after customizing it. This will enable them to add it to their predefined block library for future reuse.

#### Directory:
**`packages/editor/src/js/ext`**

#### Steps:

1. **Modal for Saving Block Attributes**: create a new file save-block-modal based on the other modals in the editor, for example: `packages/editor/src/js/vue/components/send-test/test-modal.js`
   - Path: `packages/editor/src/js/ext/save-block-modal.js`
   - Task:
     - Upon clicking the save button, a modal will pop up.
     - The modal will have two text input fields:
       * Name (mandatory)
       * Category (optional)
     - The "Save" button within the modal will be disabled initially.

2. **Integrate Save Button in block-wysiwyg.tmpl.html**:
   - Path: `packages/editor/src/tmpl/block-wysiwyg.tmpl.html`
   - Task:
     - Add a new button with a Font Awesome icon for saving the block.
     - This button will be placed alongside other block tools like delete, duplicate, etc.

3. **Control Button Behavior in viewmodel.js**: in this file we already defined the role of the buttons duplicate and delete and here we need to define the behavior of save block button
   - Path: `packages/editor/src/js/viewmodel.js`
   - Task:
     - Define the behavior of the save button.
     - When clicked, it should trigger the modal from `save-block-modal.js`.

### Acceptance Criteria:

- A new "Save" button is present in the central block editor.
- Clicking the "Save" button opens a modal with fields for "Name" and "Category".

### Result:
**Example after saving the block:**

Save Block Example

### CODE REVIEW

File: packages/editor/src/js/vue/components/save-block-modal/save-modal.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    handleOnSubmit() {
+      // Logic to handle the submission of the block details
+      // This can be expanded upon based on your requirements
+      console.log(
+        'Block details submitted:',
+        this.blockName,
+        this.blockCategory,
+        this.blockInformation
+      );
```
Comment (yoannfleurydev) [inline packages/editor/src/js/vue/components/save-block-modal/save-modal.js:54]: Is it a wanted behavior to keep console.log?

File: packages/editor/src/js/vue/components/send-test/test-modal.js [hunk 2]
```diff
@@ -50,6 +50,7 @@ const TestModalComponent = Vue.component('TestModal', {
     openModal() {
       this.$refs.modalRef?.openModal();
     },
+
```
Comment (yoannfleurydev) [inline packages/editor/src/js/vue/components/send-test/test-modal.js:53]: Be careful with those empty lines diff.

File: packages/editor/src/js/viewmodel.js [hunk 3]
```diff
@@ -47,8 +48,34 @@ function initializeEditor(content, blockDefs, thumbPathConverter, galleryUrl) {
     logoPath: 'rs/img/mosaico32.png',
     logoUrl: '.',
     logoAlt: 'mosaico',
+    toggleSaveBlockModal: ko.observable(null),
   };
 
+  function recursivelyUnwrapObservable(obj) {
```
Comment (yoannfleurydev) [inline packages/editor/src/js/viewmodel.js:54]: Is it possible to provide a global comment to quickly explain what observables are in knockout.js and why we need this function in the code.

File: packages/editor/src/js/vue/components/save-block-modal/save-modal.js (hunk 1 above)
Comment (omar-bear) [reply packages/editor/src/js/vue/components/save-block-modal/save-modal.js:54]: Yes, to see a feedback when i click on submit button, but i think i will remove it

File: packages/editor/src/js/vue/components/save-block-modal/save-modal.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                              placeholder="Enter block name">
+                            <label for="blockName">{{ vm.t('block-name') }}</label>
+                        </div>
+                        <div class="input-field col s12">
+                            <input
+                              id="blockCategory"
+                              v-model="blockCategory"
+                              type="text"
+                              name="blockCategory"
+                              placeholder="Enter block category (optional)">
```
Comment (FlorianGille) [inline packages/editor/src/js/vue/components/save-block-modal/save-modal.js:83]: Translation to add

File: packages/editor/src/js/vue/components/save-block-modal/save-modal.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                </div>
+                <form class="col s12">
+                    <div class="row">
+                        <div class="input-field col s12">
+                            <input
+                              id="blockName"
+                              v-model="blockName"
+                              type="text"
+                              name="blockName"
+                              placeholder="Enter block name">
```
Comment (FlorianGille) [inline packages/editor/src/js/vue/components/save-block-modal/save-modal.js:74]: Translation to add

**Comment by FlorianGille** [comment]
If you can also add the translation for the text when we are hovering the button to save (with the download icon), check the `public/lang/mosaico-fr.json` file.

**Pull request merged**